commercial/app/controllers/commercial/BookOffersController.scala

```diff
@@ -66,9 +74,15 @@ class BookOffersController(bookFinder: BookFinder, bestsellersAgent: Bestsellers
         }
     }
 
-    val isbns = request.queryString.getOrElse("t", Nil)
-    bestsellersAgent.getSpecificBooks(isbns) map { specificBooks =>
-      result(specificBooks ++ bestsellersAgent.bestsellersTargetedAt(segment))
+    booksSample(specificIds, segment) map result
+  }
+
+  def getBooks = Action.async { implicit request =>
+    booksSample(specificIds, segment) map { books =>
+      val json = Json.toJson(books)
```

Comment:
Do you need to `Json.toJson` with the recent refactor of `JsonComponent` by @JonNorman